### PR TITLE
feat: restrict intake to zones based on flag

### PR DESCRIPTION
**Pull request by rafasdc**
Implements [NDT-9] [NDT-6]

[NDT-9]: https://connectivitydivision.atlassian.net/browse/NDT-9?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[NDT-8]: https://connectivitydivision.atlassian.net/browse/NDT-8?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[NDT-6]: https://connectivitydivision.atlassian.net/browse/NDT-6?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/components/Form/ApplicationForm.tsx [hunk 1]
```diff
@@ -320,7 +343,8 @@ const ApplicationForm: React.FC<Props> = ({
         areAllAcknowledgementsChecked &&
         jsonData?.review?.acknowledgeIncomplete &&
         !isSubmitted &&
-        isEditable
+        isEditable &&
+        !isProjectAreaOpen
```
Comment (AntBush) [inline app/components/Form/ApplicationForm.tsx:347]: Should probably include in the dependency array - could cause false errors when attempting to submit

File: app/components/Form/ApplicationForm.tsx [hunk 2]
```diff
@@ -357,9 +381,22 @@ const ApplicationForm: React.FC<Props> = ({
...
+    }
+    if (isSubmitPage) {
+      updateAreAllSubmissionFieldsSet(newFormSectionData);
+    }
+    if (isProjectAreaPage) {
+      const projectAreaAccepted = acceptedProjectAreasArray.includes(
+        newFormSectionData?.geographicArea[0]?.toString()
+      );
+      setProjectAreaOpen(!projectAreaAccepted);
+      setProjectAreaModalOpen(
```
Comment (AntBush) [inline app/components/Form/ApplicationForm.tsx:395]: I'm not sure if this is intended behaviour, but right now anytime someone clicks on a form element in the project area page while a zone that isn't for this intake is chosen you get the modal popping up every time. 

Click zone 7 -> modal opens -> close modal -> click "yes" for project spanning multiple territories -> modal opens again

Comment (AntBush) [reply app/components/Form/ApplicationForm.tsx:395]: Also I don't know if this was introduced in this PR, but right now I'm only able to select 1 zone at a time.

**Review by AntBush — CHANGES_REQUESTED** [review]
Looks good and is working well! Just a bug and some small notes/nits

File: app/components/Form/ApplicationForm.tsx [hunk 3]
```diff
@@ -183,7 +184,19 @@ const ApplicationForm: React.FC<Props> = ({
...
+  const forceLatestSchema =
+    draftAppsUseLatestSchema?.value &&
+    typeof draftAppsUseLatestSchema.value === 'boolean'
+      ? draftAppsUseLatestSchema?.value
+      : null;
+  const acceptedProjectAreas = useFeature('intake_zones');
+  const acceptedProjectAreasArray =
+    typeof acceptedProjectAreas !== 'undefined' &&
+    acceptedProjectAreas?.value !== null &&
+    typeof acceptedProjectAreas?.value === 'string'
```
[the remark this answers is not in the document]
Comment (rafasdc) [reply app/components/Form/ApplicationForm.tsx:197]: yes, thanks!

File: app/components/Form/ApplicationForm.tsx [hunk 4]
```diff
@@ -261,6 +280,8 @@ const ApplicationForm: React.FC<Props> = ({
       rowId,
       finalUiSchema,
       setTemplateData,
+      isProjectAreaOpen,
+      acceptedProjectAreasArray,
```
[the remark this answers is not in the document]
Comment (rafasdc) [reply app/components/Form/ApplicationForm.tsx:282]: From what I tested the acceptedProjectAreas and the acceptedProjectAreasArray doesn't refresh automatically from growthbook without a manual page refresh so it doesn't make an immediate difference, I'm adding for consistency sake.

File: app/components/Form/ApplicationForm.tsx (hunk 1 above)
Comment (rafasdc) [reply app/components/Form/ApplicationForm.tsx:345]: yup, great catch!

File: app/components/Form/ApplicationForm.tsx (hunk 2 above)
Comment (rafasdc) [reply app/components/Form/ApplicationForm.tsx:395]: great catch, working on the fix. 

as for the second question, yes, that was handled as part of a different ticket. new intakes will only allow you to select one project area at a time.

Comment (AntBush) [reply app/components/Form/ApplicationForm.tsx:395]: Ahh gotcha

**Review by AntBush — APPROVED** [review]
⭐⭐⭐⭐⭐

**Pull request merged**